beacon-chain/blockchain/process_block.go

```diff
@@ -163,7 +159,29 @@ func (s *Service) postBlockProcess(ctx context.Context, signed interfaces.ReadOn
 			Optimistic:  optimistic,
 		},
 	})
-
+	if blockRoot == headRoot && isEarly {
```

Comment:
Why is this a separate condition instead of placing it after #L118? Does locking make this difficult?

Reply:
+1 why not use it in the initial condition above ? Would make it clearer to understand. Maybe having it in a separate method would be cleaner